openstack/containerinfra/v1/nodegroups/doc.go

```diff
@@ -0,0 +1,68 @@
+/*
+Package nodegroups provides methods for interacting with the Magnum node group API.
+
+All node group actions must be performed on a specific cluster,
+so the cluster UUID/name is required as a parameter in each method.
+
+
+Create a client to use:
+
+    opts, err := openstack.AuthOptionsFromEnv()
+    if err != nil {
+        panic(err)
+    }
+
+    provider, err := openstack.AuthenticatedClient(opts)
+    if err != nil {
+        panic(err)
+    }
+
+    client, err := openstack.NewContainerInfraV1(provider, gophercloud.EndpointOpts{Region: os.Getenv("OS_REGION_NAME")})
+    if err != nil {
+        panic(err)
+    }
+
+    client.Microversion = "1.9"
+
+
+Example of Getting a node group:
+
+    ng, err := nodegroups.Get(client, clusterUUID, nodeGroupUUID).Extract()
+    if err != nil {
+    	panic(err)
+    }
+    fmt.Printf("%#v\n", ng)
+
+
+Example of Listing node groups:
+
+    // Options:
+    // Marker:  pagination marker for large data sets. (UUID field from node group).
+    // Limit:   maximum number of resources to return in a single result.
+    // SortKey: column to sort results by. Default: id.
+    // SortDir: direction to sort. "asc" or "desc". Default: asc.
+    // Role:    list all nodegroups with the specified role.
```

Comment:
nit: The above docs can be removed. The ListOpts struct with comments will be formatted in a readable way in godoc.

Reply:
Done.